### PR TITLE
Add the `logs` subcommand to the CLI

### LINKED ISSUE #114: Implement a sub-command to show or follow the Elastic Agent logs
Debugging Elastic Agent is currently not as easy as it should be. In case of issue, the right paths for the logs have to be found and read one by one. It would be very convenient if elastic agent would offer a command to get the logs and metrics.

To tail all the logs, something like the following would be useful:

```
elastic-agent logs -f
```

Maybe later support for filtering logs from only a specific process could be added. One step further would be, that on the fly the logging level could be changed.

The same is true for metrics. Would be nice of a snapshot of the metrics could be gathered with something like the following:

```
elastic-agent metrics
```

### PR DESCRIPTION
**Pull request by rdner**
## What does this PR do?

It adds the `logs` subcommand.

Following is supported:

* Print the last `-n` lines of the logs (10 by default)
* Follow the logs after printing
* Filter logs by a component ID

## Why is it important?

Improves development and troubleshooting experience.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
~~- [ ] I have added an integration test or an E2E test~~

## How to test this PR locally

* Unit tests with a very good coverage (all the successful paths and edge cases are covered)
* Install the agent built from this branch and then use `elastic-agent logs -f`

## Related issues

- Fixes #114

## Screenshots

https://github.com/elastic/elastic-agent/assets/887952/d74fe794-25e6-493b-8045-6539e048f761

### CODE REVIEW

**Review by pchila — APPROVED** [review]
Just an optional small suggestion, otherwise LGTM

File: internal/pkg/agent/cmd/logs.go [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+	lines, _ := cmd.Flags().GetInt("number")
+	follow, _ := cmd.Flags().GetBool("follow")
+
+	var filter filterFunc
+
+	if component != "" {
+		filter = createComponentFilter(component)
+	}
+
+	logsDir := filepath.Join(paths.Home(), logger.DefaultLogDirectory)
```
[the remark this answers is not in the document]
Comment (rdner) [reply internal/pkg/agent/cmd/logs.go:145]: According to @blakerouse the current location is the right location to watch for logs.

**Comment by pchila** [comment]
Suggestion for future dev: could we add a `--color` option to have colored output (using something like https://github.com/fatih/color) for the different log levels? It could be useful to spot at a glance error/warning logs mixed with other log levels ;)

**Comment by michalpristas** [comment]
tested on windows it works, all flags tested
